### DISCUSSION TITLE
Can you please integrate D2-01-09

### DISCUSSION THREAD

**Discussion by chma73** [mak-gitdev/HA_enoceanmqtt#79] [Request]
Hi, can you please integrate D2-01-09 that would be really helpful for me. It's this device Permundo Smart Plug PSC234. Thanks

**Comment by mak-gitdev** [comment]
https://github.com/users/mak-gitdev/projects/1?pane=issue&itemId=35565686

**Comment by mak-gitdev** [comment]
[Done](https://github.com/users/mak-gitdev/projects/1?pane=issue&itemId=35565686) in [0.1.27](https://github.com/mak-gitdev/HA_enoceanmqtt/commit/0d0a4cdb3d74b191feb0cc9169e11f10243d1ec9)

**Comment by onedread23** [comment]
Thx for the integration, i have added it to HA and if i press the button on the device it self i see the change in HA but i can not turn it on or off via HA.

What i am doing wrong?

[Powerplug/OfficeDesk]
address         = 0x01928EFE
rorg            = 0xD2
func            = 0x01
type            = 0x09

**Reply by mak-gitdev** [reply]
Hi @onedread23, 

Did you pair you EnOcean transceiver to your D2-01-09 device ?
The pairing procedure is described [here](https://github.com/mak-gitdev/HA_enoceanmqtt/wiki/Pairing-your-device).

**Reply by onedread23** [reply]
My devices are already paired with my stick i just moved the stick over from FHEM to Homeassistant so i believe the pairing got not changed.

**Reply by mak-gitdev** [reply]
@onedread23 

I don't know how FHEM handles pairing so the paired address might be different between enoceanmqtt and FHEM.

I suggest to reset your device and pair your device as mentioned in the link.

**Reply by onedread23** [reply]
ok thats definiltly worth a try. One more question do i have to set the sender address in the config file or will that be done automatically?

But one thing if my devices would not be paired with my stick would i than see the status change in HA when i press the button on the powerplug?

I have also a rocker switch where i do see the pressed buttons in HA

![image](https://github.com/mak-gitdev/HA_enoceanmqtt/assets/35652798/67709333-f64c-4155-a0aa-c6545735863a)

**Reply by mak-gitdev** [reply]
@onedread23,

> ok thats definiltly worth a try. One more question do i have to set the sender address in the config file or will that be done automatically?

No don't add a sender ID. The base ID of your EnOcean transceiver will be used.

> But one thing if my devices would not be paired with my stick would i than see the status change in HA when i press the button on the powerplug?

Yes, status messages are broadcasted so every EnOcean device with RX capabilities can "see" them.

> I have also a rocker switch where i do see the pressed buttons in HA

You should have binary sensors named:
- binary_sensor.e2m_switches_officedesk_ai_pressed
- binary_sensor.e2m_switches_officedesk_ao_pressed
- binary_sensor.e2m_switches_officedesk_bi_pressed
- binary_sensor.e2m_switches_officedesk_bo_pressed
- binary_sensor.e2m_switches_officedesk_pressed

**Reply by onedread23** [reply]
Thank you sir, the new pairing did resolve my problem.

Thx for your effort.
cheers

**Reply by mak-gitdev** [reply]
@onedread23,

Perfect then :+1:
